Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/device/DefaultSystemInfoProvider.kt

```diff
@@ -1,19 +1,23 @@
 package de.rki.coronawarnapp.util.device
 
 import android.content.Context
-import android.content.res.Resources
 import android.os.Build
+import de.rki.coronawarnapp.util.di.AppContext
 import java.util.Locale
 import javax.inject.Inject
+import javax.inject.Singleton
 
-class DefaultSystemInfoProvider @Inject constructor(context: Context) : SystemInfoProvider {
+@Singleton
+class DefaultSystemInfoProvider @Inject constructor(
```

Comment:
I would prefer an extension function here.

Something like

```kotlin
val Context.locale ...
// or
fun Context.getLocale
```

the system info provider is IMHO overkill and also tied to the application context while `context.resources` is affected by theming  and configuration changes, so it could differ from activity to activity.

Reply:
Fair point, done.